16.0 mig commown devices

9184Lucky opened the pull request.

fcayre committed:
Update location on create move lines in internal_picking

Also update the qty_done on lines instead of the computed field
quantity_done on the move itself.

fcayre committed:
Add picking action to force the effective date of a picking and its moves to the picking's scheduled date

fcayre committed:
fr t10n update

fcayre committed:
Set scheduled date on picking created by our custom wizards

github-grap-bot committed:
commown_devices 12.0.1.0.1

fcayre committed:
First revision with fees modelisation and computation

Computations are run using a dedicated button and change their state
accordingly. Once done, the computation can be reset to draft. This is
required if you want to modify the fees definition, which cannot be
changed if there is at least one non-draft computation.

fcayre committed:
Fix small code style oddities

Commit:
Add the contract_id field in the picking form view

github-grap-bot committed:
commown_devices 12.0.1.0.2

github-grap-bot committed:
commown_devices 12.0.1.0.3

fcayre committed:
Add mecanism to check a picking is present when a lead enters certain columns

... columns which name contain the '[stock: check-has-picking]' string.

fcayre committed:
fr t10n update

github-grap-bot committed:
commown_devices 12.0.1.0.4

fcayre committed:
Fixup fr t10n erroneous removal in commown_devices

github-grap-bot committed:
commown_devices 12.0.1.0.5

Commit:
Modify domain for storable product computation

Commit:
Added a field for current location and changed domain to exclude current location

Commit:
i18n + l10n-fr

Commit:
Adding tests for picking involved device wizzard

fcayre committed:
commown_device 12.0.1.0.6

Commit:
Added warning and constrains to tasks in resiliation project.

Added warning to actions that should not be used in resiliation project
and constrains to related contract's stocks to improve stock
reliability.

github-grap-bot committed:
commown_devices 12.0.1.0.7

github-grap-bot committed:
commown_devices 12.0.1.0.8

Commit:
Add an outward picking wizard on project task for not tracked products

Commit:
Add an inward picking wizard on project task for not tracked products

Commit:
Added tests

Commit:
i18n + l10n and fix typo

Commit:
Refactored contract so picking origin can be different than contract name

Also added a function to get task name in the form "Task-ID"

Commit:
Added a check on picking when a task enters a "check_picking_assigned" stage

Commit:
Refactoring to use constrain instead of write

Commit:
Correct xml_ids and add tests

github-grap-bot committed:
commown_devices 12.0.1.0.9

fcayre committed:
i18n + FR/DE t10n update

fcayre committed:
Move python code from template to model

Apart from improving code quality, this fixes a dependency problem:
the qweb template was using the quant_ids field, which is defined in
commown_devices, but commown_self_troubleshooting cannot depend on
commown_devices because it depends on commown_self_troubleshooting...

So we use a method override: commown_devices implements an override of
the contract.contract displayable_description method which uses the
quant_ids field.

fcayre committed:
Change contract dropdown description

Remove useless commitment end date and put devices before dates.

github-grap-bot committed:
commown_devices 12.0.1.0.10

Commit:
Modify the constain on stage ID to allow for check on recordset

Commit:
i18n + fr10n

github-grap-bot committed:
commown_devices 12.0.1.0.11

Commit:
Create functions to send lots and secondary products on the same picking

Commit:
Modify crm.lead picking wizard to send secondary products

Commit:
Add an attribute on project stage to enable picking control

The attribute is dedicated to check if a picking attached to the task's contract is in 'assigned' state when entering a stage with that attribute set to True.

Commit:
Rename project task method

fcayre committed:
commown_devices 1.1.0

Commit:
Add model describing relations between service variants and storable products

Commit:
Display service_storable_config on product template

Added service storable configuration field on product template and a view to display them.
This view allows configurations edition directly on product template
view.

Commit:
Add fields storing primary & secondary storable product on service variant

Also added a funnction to compute those fields using Service storable
configuarions.

fcayre committed:
Fix custom module dependencies

commown_self_troubleshooting now depends on commown_devices, not the other way around.
This requires moving the contract termination project creation to a lower level module
as commown_devices requires it.
It also simplifies the implementation of the displayable_description method of the
contract entity type, which can be entirely implemented in the commown_sel_troubleshooting
module instead of partly in commown_devices.

Commit:
i18n fr10n

github-grap-bot committed:
commown_devices 12.0.1.1.1

Commit:
Rewrite code tu use more generic picking function.

Merge all picking function in a more generic one, modifiy send and
receive function of contracts accordingly.

Commit:
Replace quant by lot in function send_device of contract model

Doing so makes it more coherent with function receive_device and also
link the contract to lot more than quant wich is needed in the next
refectoring.

Commit:
Unify send contract send_device function

Commit:
Unify receive_device function of contract model

Commit:
Added a stock location for modules and modify function accordingly

github-grap-bot committed:
commown_devices 12.0.1.1.2

Commit:
Allow user to choose if the non serial products have to be sent from New stock or repackaged

This is useless for the lots since their localisation is unique and
found from the quants.

Commit:
Add stock location for repackaged nonserial products

Commit:
Add a picking wizard for non serial product on project task

fcayre committed:
Raise an error when trying to scrap a device from a task where it is not set

Commit:
i18n fr.po

github-grap-bot committed:
commown_devices 12.0.1.1.3

Commit:
Add relation between lots and contracts.

The relations relies on the relation between pickings, scraps and
contracts.

Commit:
Specifiy where to look for lots and product when creating return picking

Looking for products in location that are children of the location of all partners wasn't specific enough and caused accessories to be received from another partner location.

This commit also modifies the location where lots are searched. It might help to detect errors.

github-grap-bot committed:
commown_devices 12.0.1.1.4

Commit:
Remove quants and picking from contract

Commit:
Remove unused function

Commit:
Fix an error that permitted to select a reserved product in a picking

fcayre committed:
Fix duplicated i18n message in commown_device fr.po

Commit:
Make the "Set scheduled date..." action usable on recordset

github-grap-bot committed:
[BOT] post-merge updates

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Add a wizard that alows linking po_lines to invoices_lines

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Add a wizzard to link picking's moves to purchase order lines

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Add link between contracts and move line

Also improve a test on website_sale_b2b to check contract to lot link

fcayre committed:
Fix incorrect domains of project tasks product and lot fields

They were using domain methods not called on existing tasks. They are
now using not stored computed dedicated domain fields which make it
easier to debug and more reliable.

Note the test method dedicated to the UI (prepare_ui) had to be enriched
by a mean to evaluate the domain attributes defined in the view.

fcayre committed:
Tests: turn hardcoded domain fields into a view-introspected automated list

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Add explicit test for action_scrap

Commit:
Create an action to set scrap related entities date to scrap's expected date

Commit:
Fix typo

Commit:
i18n fr10n

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Modify common picking function to return move line instead of picking

We also add the possibility to reuse a picking but this is not used for
now. It will be used later.

github-grap-bot committed:
[BOT] post-merge updates

fcayre committed:
Add a partner method to merge its locations and use it in partner merge wizard

fcayre committed:
Fix flake8-detected bug

Commit:
Move _assigned function to common as it will later be used elsewhere

fcayre committed:
Handle product move changes when an individual becomes a professional

In such a case, the old individual stock location is removed and
replaced by the new (=its company's) location.

fcayre committed:
Tiny perf improvement

fcayre committed:
Coding-style improvements (flake8 detected)

fcayre committed:
Move contract-related stock to the new location when creating an intermediate company

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Add a cron to warn a channel of late pickings

Commit:
Fix a bug on that caused the wizzard to crash

Commit:
Fix typo

Commit:
Fix error in test class inheritance

Commit:
Revert unused reuse_picking feature

Commit:
Migrate picking and scrap moves to contract

Commit:
Shorten location name on contract view

Commit:
Fix link between contract and scrap's move

When a scrap is created, it has no move. Therefore we need to store the contract on the scrap and associate it to its move when created

github-grap-bot committed:
[BOT] post-merge updates

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Add a toggle to hide move lines without lot in contract view

Commit:
Add a test to prevent scrapping an already scrapped device

Commit:
Improve move line display on contract view

Add decorations depending on line in or out destination

Commit:
Add test to improve coverage

Commit:
Remove a unused line in tests

Commit:
Clean outdated storable_product_id field on product template.

This field has been replace by the primary_storable_variant on products.
There is still a storable_product_id on project task.

github-grap-bot committed:
[BOT] post-merge updates

Commit:
i18n fr10n

fcayre committed:
Robustify the code that sets the contract/ lot relation on move validation

We now use a location hierarchy (based on their partners' hierarchy) to
circumvent problems due to partners changing from B2C to B2B or the
like.

Also, in the migration script, log an error instead of raising when move
and contract locations are inconsistent

Commit:
Fix migration script

The post-migrate script was looping on the contract's moves no matter
the move's global chronological order.

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Fix erroneous translation

Commit:
Fix error message

Commit:
Move a funtion to common file to use it later

Commit:
Add a check in crm_lead test

Commit:
Improve test coverage

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Set show_all_move_line default to True

Commit:
Add confirmation wizard when there is several unvalidated moves

Commit:
Add a button to display the parent of a move

Commit:
Add the origin of picking on move line tree view

Commit:
Fix boolean toggle to show/hide move lines

fcayre committed:
Add a wizard to give a device to an employee

We use a support ticket and a contract which partner is the employee
to mimic the way we handle customers. That way, these devices should
not be exceptions anymore and should be easier to manage.

fcayre committed:
Show contract on project task when set, even if not required

This way:
- we can configure a custom project for employee devices, with contract not required
  (because it does not exist when the task is created), but shown once the wizard was
  executed
- in other projects where a contract is not supposed to be there but really is, we
  can at least see it (and maybe removed it after investigating)

fcayre committed:
Add ability to print a label instantly when preparing a device to an employee

Commit:
Re-introduce a field that count the number of lot on a contract.

The field quant_nb was deleted when the quant were removed of the
contract but had not been replaced by a similar field.

fcayre committed:
[commown_devices] i18n update and FR t10n

fcayre committed:
Factor out a wizard 'to_employee' test mixin to ease testing in other modules

fcayre committed:
[commown_device] Add (empty) DE t10n file

github-grap-bot committed:
[BOT] post-merge updates

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Add a button to open the origin document of the operation

Commit:
Factorize a function to get the parent of a move line

Commit:
Copy the move line view from the contracts to the project tasks

The move lines displayed on project tasks are filtered so only the one
that comes from the current tasks are shown.
Also add the corresponding configuration on project.

fcayre committed:
[tests] Use internal_picking instead of obsolete test method

... which generates a warning as lot is on move lines, not moves.

Commit:
Remove erroneous @onchange

Commit:
i18n l10n fr

fcayre committed:
Fix incorrect context passed to validate scrap

There was a warning about it.

This was not properly tested but now it is.
Note it worked before the patch for some reason.

fcayre committed:
Avoid a warning when domain is invalid

When a location has no partner, passing it to has_partner_child_of
returns False. This now does the same without a warning like
'Unexpected domain [%s], interpreted as False'

Commit:
Factor scrap creation in setUp method

fcayre committed:
[fr t10n] Fix typos

github-grap-bot committed:
[BOT] post-merge updates

github-grap-bot committed:
[BOT] post-merge updates

Commit:
Add a delta between the two contract transfer's pickings

fcayre committed:
Add a contract stock ownership field valued "internal" or "customer"

The values are similar to the customer location usage.
This way products send to a customer in the context of a contract is sent
to a location with the right usage: internal for rentals, customer for sales.

github-grap-bot committed:
[BOT] post-merge updates

fcayre committed:
Fix one fr t10n

Commit:
Add a default grade to purchase order

The default grade is then to be set on the created lots.

Commit:
Reformating: Use recordset instead of list of lots

Commit:
Check lots to be sent are graded

Commit:
Make test name more explicit

Commit:
Remove grade on lot return

Commit:
i18n fr10n

fcayre committed:
Test fixes

In commown_devices tests, the po was using the default picking_type_id
leading to devices arriving in the sale stock.

In rental_fees tests, the repackage step was missing but was not
crashing because the repair location was a son of the default
send_device origin location (which was stock.stock_location_stock at
that time).

github-grap-bot committed:
[BOT] post-merge updates

fcayre committed:
Add missing i18n pot files

fcayre committed:
Add a global rental-dedicated stock location

fcayre committed:
Add rental-related picking types

fcayre committed:
Remove useless import in tests

fcayre committed:
Add rental-related account fields on product and product category

These fields play the same role as their sale-related equivalents, defined in the
standard account add-on: be used on purchase invoice lines, but when the invoiced
products are to be rented, not sold.

fcayre committed:
Implement invoice line account choice for rental dedicated purchased products

The expense account for products to be rented is defined on the product
or its category, and is used when generating an invoice from a purchase
order.

fcayre committed:
Customize product and product category expense account fields' string

... so that there is a clear distinction between expense accounts aimed
at the resale and rental activities.

fcayre committed:
i18n update + complete FR t10n

fcayre committed:
Improve purchase order form view

We need to see the picking type so move it in the upper, always visible
form part. Adapt another field position accordingly.

fcayre committed:
Create a customer location on sale with serial products confirmation

The location belongs to the company and is associated with the sale's shipping partner.

fcayre committed:
[tests] Fix test crash on CI env

The crash is due to website_sale_affiliate module using the werkzeug request object
in an unsuitable environment.

fcayre committed:
Fix typo in FR t10n

fcayre committed:
Add an action to add storable products on sales with sale-with-contract service products

The added storable products are the one configured on the sold
service(s) that must be a product of type "service" and a contract of
"customer" stock_ownership as a property_contract_template_id.

fcayre committed:
Inactivate picking creation on sales that have a line with a contract product

fcayre committed:
Add ability to send devices part of a sale service from a project task

For rental, we use crm leads to first study the risks and validate the
rental.

For sale with services, we use project tasks to prepare the devices, as
we do not need a risk analysis once the sale is confirmed.

The wizard to send the devices is now common to both use cases.

fcayre committed:
Link resale with services pickings to the customer order

... and its moves to the order lines.

This makes the pickings and the delivered quantities visible on the sale order.

fcayre committed:
Fix warning on odoo startup related wizard's all_products field

Namely: "Field [...].all_products depends on non-stored field product.product.type, this operation is sub-optimal"

So use the product template's type instead.

fcayre committed:
Coding style improvements

fcayre committed:
Add picking validation and contract start delivery actions to project tasks

The code was moved from crm.lead specific methods to a mixin inherited by project.task too.

A part of this code was moved from the commown module to commown_devices.

fcayre committed:
Improve existing test readability a bit

fcayre committed:
Refactor test a bit before adding a new one

fcayre committed:
Small test base class improvement to enlarge its usage

In an upcoming test we need to customize a sale order (or its products) before
it is confirmed. A lot of tests use the result of the sale confirmation so we
do not remove it but make it optional and ON by default.

fcayre committed:
Improve test coverage

fcayre committed:
Refactor picking to customer wizard test before adding one dedicated to the tasks

fcayre committed:
Refactor crm lead picking test to ease test code sharing

fcayre committed:
Add a wizard to send sold devices to customers from project tasks

fcayre committed:
i18n update (pot files only)

fcayre committed:
FR t10n update

fcayre committed:
Automatically rename fields with script

Script is:
```

rename() {
    oldword=$1
    newword=$2

    git grep -l $oldword | grep -vE '\.po(t)?$' | xargs sed -i "s@$oldword@$newword@g"
}

rename is_rental has_recurrent_payment
rename rental_price_extra recurrent_payment_amount_extra
rename rental_price recurrent_payment_amount
rename rental_tax_ids recurrent_payment_tax_ids
rename rental_frequency recurrent_payment_frequency
rename rental-ratio recurrent-payment-ratio
```

fcayre committed:
i18n update of all modules impacted by the field renamings

fcayre committed:
Post-merge updates

fcayre committed:
Merge views which only difference was their concrete model

We can use a single view for both crm.lead and project.task
to customer picking wizards, so merge them to simplify things.

fcayre committed:
Fix wrong `all_products` permissions for customer picking wizard view

The view with the permission restriction is the one to be
returned as Odoo manages to apply the restriction only when
applicable (=when current user is in the restriction group).

Before the patch we were returning the view without group
restriction, which obviously gives the same result for all
users, which is wrong.

github-grap-bot committed:
[BOT] post-merge updates

Commit:
[IMP] commown_devices: black, isort, prettier

Commit:
[MIG] commown_devices: Migration to 16.0

Commit:
[test] Match account.account changes in account method

- We replaced user_type_id by account_type
- We removed thedeprecated internal_type field

fcayre committed:
Allow a subcompany as a destination partner for the to-employee wizard

Commit:
Replace get_invoice_line_account() by _compute_account_id()

I made a specific commit for this change because I am not certain of
the way I've done it and wanted it to be more visible and easier to revert.

Commit:
Add a AbstractModel to ToCustomerPickingMixin class

Commit:
Change move_lines fields of stock.picking to move_ids

A bit of work but it was one of the dumbest field name choice ever made
so happy to do it.

Commit:
Remove deprecated public field on mail channel

Commit:
Remove deprectated binding_model field

binding_model_id is already defined in those records

Commit:
Recreate partner_id field on stock_location

This field has been deleted from odoo core but we rely quite a lot on it.
We decided to re-create it for the migration and put a ticket to replace it after
the migration (id: 38173)

Commit:
Add sequence code that will be used if not sequence id is found

In our case it won't append but the field is required.
See its usage in /odoo/addons/stock/models/stock_picking.py +113

Commit:
Adapt to new project_project view architecture

Commit:
Change the way fields are declared differently for groups of users

Commit:
Replace AccountInvoice by AccountMove

Also replace AccountInvoiceLine by AccountMoveLine

Commit:
Add missing _description and access rules to wizards

Access rules should be defined more specificly depending on groups but
it is not a work to do during migration. I added rules equivalent to
current access.

Commit:
[test] Replaced SavepointCase class by TransactionCase

Commit:
[test] Replaces at_install/post_install decorators by tagged

Commit:
[test] Remove useless class dependency

Commit:
[test] Replaced setUp methods by setUpClass

Some of the methods used in the set-up of the classes are declared in product_rental, and the setUpClass class method can't use instance methods, we modify a few used product_rental instance methods to be class methods.

Commit:
[test] Inserted flush for computations and updates to the data

I don't specifically understand why it makes the tests pass, but the tests fail otherwise.

Commit:
Adapt to new product variant modelisation

Commit:
Adapt to changes in stock.move model

Commit:
Inverse the order of the super() call so account_id is not overriden

Commit:
[test] Changed inventories initialisation in tests

Inventories are now made directly from quants.

Commit:
Make _action_done() override more resiliant to signature changes

Commit:
Remove lasts warning in logs

Commit:
Change removed action_done() method on picking by button_validate()

Commit:
[test] Adapt to changes in stock.move.line model

Commit:
Migrated templated from Jinja to QWeb

Commit:
Deleted references to removed scrap location

The stock module changed the way various locations (incl. scrap) are generated - rather than having a general scrap location, the module creates a location for each company in the environnement.
However, we don't need to set a new default scrap location, since the stock.scrap creation sets a default scrap location, which is either the default company or the environnement company.

Commit:
[test] Factorized common test code

Commit:
[test] Changed demo data retrival from product_rental

Commit:
[test] Removed useless attribution

_origin is a BaseModel method marked as a property (with the eponymous decorator) - however, this property has no setter method, and so this is an invalid operation

Commit:
WARNING This is a fix. We do not have a better solution at the moment.

The function is called by an onchange. When creating the record, self is
not defined (NewId state) which makes the database call (quant.search)
fails.

Commit:
[test] Fixed  prepare_ui method

- We added two try/except blocks :
* one around the domain evaluation to make it more robust.
* one around the field retrival, in case a field is present in the view, but not in created_model (for instance, a field from a related model)
- In the search of possible values, we use a copy of the domain.
This is because odoo modifies the domain when we call search. This causes weird bugs where domains of other fields were modified.
- We added the uid in the context before calling safe_eval, which brings the domain search closer to the real behaviour.

This is necessary when there are fields that are present in the view but
not in the created_model (e.g. field of a related model that appears on
the view)

Commit:
Re-define move_line_ids on project.task

I do not recall the reasons behind the choice of a computed method here
but it seems to be a mistake. Related field are made for this exact
purpose.

Commit:
[test] Fix project.task test by making assertion more robust

Commit:
Prefix of purchase order sequence have changed for "P".

This makes this implementation even weaker.

Commit:
[test] Adapted to changes of the account.account model

Commit:
[test] Adapted products retrival in test_ui

This is mainly due to changes with defaults

Commit:
[test] Add a flush before cache invalidation

Commit:
[test] Remove useless assertRaises

The checked operation no longer raises a ValidationError exception, which was potentially raised in Odoo v12

Commit:
Remove deprecated function

stock.move.line name is now calculated via _compute_name which depends
on product_id

Commit:
[test] Change external id of demo project.task record

Commit:
[test] Remove deprecated date_expected field on stock.scrap creation

Commit:
Change mechanism to set a date on stock.scrap

date_expected field has been removed from stock.scrap.
But stock.move.line date is now writable.
Therefore we propose a new way of setting the date of a scrap and its
move (to be able to get a correct traceability). The user has to modify
the date of the move line and then use this action.

We could discuss the relevance of keeping the action and not making it a
depends on move_id.stock_move_line.date.

Commit:
[test] fields_view_get has been deprecated for get_view

Commit:
[test] Changes exception.name to exception.args[0]

Commit:
Set the date_start of the contract when calling _modify_from_description

I do not understand why it is necessary now, I have to dig it but it is
a fields that we declare so I don't know why it should be updated
automaticaly.

Honeyxilia commented on commown_devices/models/account_invoice_line.py:
```diff
@@ -1,20 +0,0 @@
-from odoo import models
```
Est-ce que tu avais utilisé `git mv` pour renommer le fichier ? Sinon, je pense qu'il serait bien de le faire ici pour l'historique et le suivi c:

9184Lucky replied:
Yes j'avais utilisé git mv mais je pense que y'a trop de différence entre les deux fichiers pour que il comprenne que c'est les mêmes....

Honeyxilia replied:
Okie c:

Honeyxilia commented on commown_devices/models/wizard_project_task_to_employee.py:
```diff
@@ -62,11 +62,11 @@ def execute(self):
-                    ("id", "=", self.task_id.partner_id.id),
-                ]
-            ):
+            [
```
remarque très mineure, mais je pense que ceci pourrait être fusionné au commit black, flake8

Honeyxilia commented on commown_devices/__manifest__.py:
```diff
@@ -43,6 +43,7 @@
         "views/project_task_type.xml",
```
dans le message du commit, il y a une typo : 
- but *be* rely -> but *we* rely

Honeyxilia commented on commown_devices/data/stock_picking_type.xml:
```diff
@@ -22,6 +22,7 @@
     <field name="sequence">100</field>
```
erreur dans le message du commit : 
- it will not *happend* but -> it will not *append* but

Aussi, pour rendre le message plus idiomatique : 
- it *will not* -> it *won't*

Honeyxilia commented on commown_devices/tests/test_account_move_line.py:
```diff
@@ -54,15 +54,16 @@ def purchase(self, picking_type_ref):
         invoice = self.env["account.move"].create(
             {
-                "type": "in_invoice",
+                "move_type": "in_invoice",
```
Je pense que cette ligne aurait plus sa place dans ce commit où tu fais les changements de champs liés à `account.move` 
https://github.com/commown/commown-odoo-addons/pull/405/commits/4a85730a9fae09d0920a3bb6745d7b83b65d47b2

Honeyxilia commented on commown_devices/tests/test_account_move_line.py:
```diff
@@ -54,15 +54,16 @@ def purchase(self, picking_type_ref):
     def invoice_account(self, po):
```
Quelques petites erreurs de message de commit (bff1290) : 
- purchase_order_change() *methode does not longer exist* -> puchase_order_change() *method no longer exists*
- purchase_id *have* to -> purchase_id *has* to
- extra lines are *generates* -> extra lines are *generated*

Honeyxilia commented on commown_devices/data/stock_location.xml:
```diff
@@ -61,6 +61,13 @@
     <field name="location_id" ref="stock.stock_location_suppliers" />
   </record>
 
+  <record id="stock_location_scrap" model="stock.location">
```
D'après le code de `stock.location` (méthode `create_missing_scrap_location`, qui est utilisé dans `stock_data.xml`), il y a deux emplacement scrap utilisé pour les companies de l'environnement, est-ce qu'elles ne correspondent pas au besoin ?

Honeyxilia commented on commown_devices/tests/test_crm_lead.py:
```diff
@@ -1,7 +1,7 @@
 from datetime import date
```
Petite erreur dans le message de commit : 
- to v12 raise mecanisme -> to *a* V12 raise *mechanism*

Commit:
[test] Made account.move.line tests more robust to multi-company environments

Commit:
Change StockProductionLot to StockLot

Commit:
[test] Fix invoice_account test method

- The purchase_order_change() method no longer exists and seems to have been replaced by _onchange_purchase_auto_complete()
- purchase_id has to be declared after the creation of the account.move record.
I don't fully understand why, but the purchase_id field is not set otherwise.
- I added a filter on the lines to only return concerning products, because extra non-product lines are inserted by _sync_dynamic_lines(). I don't know if this is expected behaviour.

Please pay extra attention reading this commit.
This commit and the preceding ones make the test_account_move_line.py tests pass.

fcayre commented on commown_devices/models/common.py:
```diff
@@ -251,7 +251,9 @@ def _set_date(entity, value, attr_name):
 
 
-class ToCustomerPickingMixin:
+class ToCustomerPickingMixin(models.AbstractModel):
```
Je ne comprends pas ce changement : si ça devient un modèle (abstrait), alors il faut en hériter avec _inherit, pas avec un héritage python. Cf. par exemple mail.thread.
Il y a sans doute un truc qui m'échappe, mais alors ça mériterait de figurer dans le message de commit.

fcayre commented on commown_devices/tests/common.py:
```diff
@@ -433,8 +435,16 @@ def setUpClass(cls):
-                ("attribute_value_ids", "ilike", with_usbc.id),
+                (
+                    "product_template_variant_value_ids.product_attribute_value_id",
+                    "ilike",
```
Je ne comprends pas comment "ilike" peut fonctionner ici... "=" serait très bien non ?

fcayre commented on commown_devices/tests/common.py:
```diff
@@ -433,8 +435,16 @@ def setUpClass(cls):
+                ),
+                (
+                    "product_template_variant_value_ids.product_attribute_value_id",
+                    "ilike",
```
idem, et plus loin aussi.

Commit:
Simplify _compute_origin() and add tests

Commit:
Add tests to improve coverage

We also removed an useless "grade_lot=False" argument in an adjust_stock
call.

Commit:
[test] Add tests to improve coverage

We also removed an useless "grade_lot=False" argument in an adjust_stock call.

Commit:
Change the way dynamic domain are declared.

The way it was done before was not working in the interface

fcayre commented:
/ocabot rebase

9184Lucky commented:
/ocabot merge patch
